registry.go

```diff
@@ -108,3 +112,9 @@ type SignatureService interface {
 	// Put stores the signature for the provided digest.
 	Put(dgst digest.Digest, signatures ...[]byte) error
 }
+
+// CatalogService provides a way of retrieving the names of each of the repositories
+type CatalogService interface {
+	// Get retrieves repository names from the registry.
+	Get(n int, q string) (p []string, moreEntries bool, err error)
```

Comment:
After looking at this PR, this might be better as `Repositories` or something more descriptive than `Get`. We may not even need the separate `CatalogService` object, since that complements `Repository`. It would just have to return an error.

Put more clearly, this may be fine as the following:

```
type Namespace interface {
  //... other methods

  Repositories(n int, q string) (p []string, moreEntries bool, err error)
}
```